integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.security.sasl.AuthenticationException;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import tachyon.Constants;
+import tachyon.client.FileSystemMasterClient;
+import tachyon.master.LocalTachyonCluster;
+import tachyon.security.authentication.AuthenticationFactory.AuthType;
+import tachyon.security.authentication.AuthenticationProvider;
+
+/**
+ * Though its name indicates that it provides the tests for Tachyon authentication. This class is
+ * likely to test four authentication modes: NOSASL, SIMPLE, CUSTOM, KERBEROS.
+ */
+public class MasterClientAuthenticationIntegrationTest {
+  private LocalTachyonCluster mLocalTachyonCluster = null;
```

Comment:
Are there tests that authenticate with LocalTachyonClusterMultiMaster (tachyon in fault tolerant mode)?